internal/handlers/types.go

```diff
@@ -0,0 +1,33 @@
+package handlers
+
+import (
+	"github.com/repriest/url-shortener/internal/config"
+	"github.com/repriest/url-shortener/internal/storage"
+)
+
+type Handler struct {
+	cfg *config.Config
+	st  *storage.Repository
+}
+
+func NewHandler(cfg *config.Config, st *storage.Repository) *Handler {
```

Comment:
Было бы здорово реализовать зависимость от Storage через interface:
```go
type Storage interface {
	Load() ([]URLEntry, error)
	Append(entry URLEntry) error
	BatchAppend(entries []URLEntry) error
	Close() error
}
```
```suggestion
func NewHandler(cfg *config.Config, st *Storage) *Handler {
```